src/document.rs

```diff
@@ -70,6 +160,88 @@ impl Document {
         }
     }
 
+    fn sort_nodes(&self, nodes_orig: &Vec<Node>) -> Vec<Node> {
+        if nodes_orig.len() < 2 {
+            return nodes_orig.clone();
+        }
+
+        let nodes = nodes_orig.clone();
+
+        let mut pre = vec![];
+        let mut elems = vec![];
+        let mut has_text = false;
+        let mut post = vec![];
+        for node in &nodes {
+            match node {
+                Node::Element(elem) => {
+                    elems.push(ElementAndContext {
+                        pre: pre.clone(),
+                        elem: node,
+                        attrs: self.attrs.get(elem.0).unwrap(),
+                        elem_val: self.nodes.get(elem.0).unwrap().as_element().unwrap(),
+                    });
+                }
+                Node::Text(_) => {
+                    has_text = true;
+                    post.push(node);
+                }
+                Node::CDataSection(_) => pre.push(node),
+                Node::ProcessingInstruction(_) => post.push(node),
+                Node::Comment(_) => pre.push(node),
+                Node::DocumentType(_) => {
+                    panic!("DocumentType found inside an element; that should never be the case")
+                }
+            }
+        }
+        if has_text && !elems.is_empty() {
+            println!(
+                "Element contains both element(s) and text; \
+            we thus consider all elements to be inline-elements \
+            (think of the *bold* element in HTML), and thus we will not sort them."
+            );
```

Comment:
This should be logging not a println; if we don't already have the `tracing` dependency, let's add it.